tests/app/dao/test_provider_details_dao.py

```diff
@@ -155,75 +154,6 @@ def set_primary_sms_provider(identifier):
     dao_update_provider_details(secondary_provider)
 
 
-@pytest.mark.serial
```

Comment:
This is related to the `client_to_use` method, and how the `PROVIDER_STRATEGIES_ENABLED` feature flag makes hitting this code impossible. These tests were covering that unreachable code.
![image](https://github.com/user-attachments/assets/8518ac2b-94b5-4a43-87e5-9d28e41c6c9b)